packages/fhir-converter/src/lambdas/ccda-to-fhir.js

```diff
@@ -0,0 +1,192 @@
+var path = require("path");
+var fs = require("fs");
+var Promise = require("promise");
+var compileCache = require("memory-cache");
+var constants = require("./lib/constants/constants");
+var errorCodes = require("./lib/error/error").errorCodes;
+var errorMessage = require("./lib/error/error").errorMessage;
+var HandlebarsConverter = require("./lib/handlebars-converter/handlebars-converter");
+var dataHandlerFactory = require("./lib/dataHandler/dataHandlerFactory");
+var {
+  extractEncounterTimePeriod,
+  getEncompassingEncounterId,
+} = require("./lib/inputProcessor/dateProcessor");
+
+const { createNamespace } = require("cls-hooked");
+var session = createNamespace(constants.CLS_NAMESPACE);
+
+var rebuildCache = true;
+
+function GetHandlebarsInstance(dataTypeHandler, templatesMap) {
+  // New instance should be created when using templatesMap
+  let needToUseMap =
+    templatesMap && Object.entries(templatesMap).length > 0 && templatesMap.constructor === Object;
+  var instance = HandlebarsConverter.instance(
+    needToUseMap ? true : rebuildCache,
+    dataTypeHandler,
+    path.join(constants.BASE_TEMPLATE_FILES_LOCATION, dataTypeHandler.dataType),
+    templatesMap
+  );
+  rebuildCache = needToUseMap ? true : false; // New instance should be created also after templatesMap usage
+
+  return instance;
+}
+
+function generateResult(
+  dataTypeHandler,
+  dataContext,
+  template,
+  patientId,
+  encounterTimePeriod,
+  encompassingEncounterId
+) {
+  var result = dataTypeHandler.postProcessResult(
+    template(dataContext, {
+      data: { metriportPatientId: patientId, encounterTimePeriod, encompassingEncounterId },
+    })
+  );
+  return Object.assign(dataTypeHandler.getConversionResultMetadata(dataContext.msg), {
+    fhirResource: result,
+  });
+}
+
+async function ccdaToFhir(ccda, patientId) {
+  return new Promise((fulfill, reject) => {
```

Comment:
It would be good to move this to regular await like code - the function is already `async`, no need to wrap the code inside a promise.

Reply:
trying to keep the logic as similar as possible ahead of a refactor for now - this main body of logic is also in the fhir converter